frontend/model/notifications/utils.js

```diff
@@ -66,36 +66,44 @@ export function applyStorageRules (notifications: Notification[]): Notification[
     items.length = MAX_COUNT
   }
   // Sort items in chronological order, newest items first.
-  // In case two items have the same age, then make the higher priority item come first.
-  return items.sort((itemA, itemB) => age(itemA) - age(itemB))
+  return items.sort(compareOnAge)
 }
 
-export function compareOnPriority (notificationA: Notification, notificationB: Notification): -1 | 0 | 1 {
+// Compares notifications, so that newer ones come first.
+export function compareOnAge (notificationA: Notification, notificationB: Notification): -1 | 1 {
+  return age(notificationA) <= age(notificationB) ? -1 : 1
+}
+
+// Compares notifications, so that higher priority ones come first.
+export function compareOnPriority (notificationA: Notification, notificationB: Notification): -1 | 1 {
   const TWO_DAYS = 2 * ONE_DAY
   const FIFTEEN_DAYS = 15 * ONE_DAY
   const THREE_WEEKS = 21 * ONE_DAY
   const preferA = -1
   const preferB = 1
 
-  if (notificationA.read && !notificationB.read) {
-    if (age(notificationA) < ONE_DAY && age(notificationB) > FIFTEEN_DAYS) {
-      return preferA
-    }
-    if (age(notificationA) < TWO_DAYS && age(notificationB) > THREE_WEEKS) {
-      return preferA
-    }
-    return preferB
-  }
-  if (!notificationA.read && notificationB.read) {
-    if (age(notificationA) > FIFTEEN_DAYS && age(notificationB) < ONE_DAY) {
-      return preferB
-    }
-    if (age(notificationA) > THREE_WEEKS && age(notificationB) < TWO_DAYS) {
-      return preferB
+  // If the given notifications are both read or both unread, then prefer the newer one.
+  if ((notificationA.read && notificationB.read) || (!notificationA.read && !notificationB.read)) {
```

Comment:
This check is intended to prevent any duplicate references to the same notifification object to end up in the state Array.

Now you're right that two distinct notification objects bearing the same data would pass this check, but they would have been created through separate calls to `gi.notifications/emit`, so I think that's fine. Moreover this is quite unlikely to happen since a timestamp is included in the data, so both calls would have to occur in the same millisecond.

However, now that I'm thinking about it thanks to your input, maybe an extra check would be useful to make sure we don't issue equivalent notifications during the same millisecond or during a short time interval.

Reply:
But you are only comparing `boolean` typed `read` values here.

Reply:
I was in fact replying to your previous comment above, regarding `state.includes(notification)`; for some reason no text field showed up there so I had to reply in the next available field.

As for the latter comment, after thinking about it I think that indeed `if (notificationA.read === notificationB.read)` turns out to be the same, but it wasn't obvious at all for me. So the longer version is likely easier to understand.

Reply:
@snowteamer. what I am concerned about is that in my opinion `state.includes(notification)` will always be false unless we find a notification from the state itself and just to check using the `includes` prototype function. Because they don't compare values, they compare with the reference.

Reply:
@Silver-IT Yea it is supposed to always be false, unless we later change the code in a dubious way. That's the idea. It also gives a clue that `state` is logically an ordered set of objects, although being implemented as an array. However I will remove it if it turns out to be too confusing.

Reply:
@snowteamer Yeah I agree with @Silver-IT's observation, if this `if` could be simplified to `if (notificationA.read === notificationB.read)` that would be preferable. Save a few CPU cycles and some battery life :)

The comment (great job on that!) makes it clear what the intent is.